### PR TITLE
Refactor view hierarchy to achieve dax dialog animation growing/shrinking

### PR DESCRIPTION
**Pull request by alessandroboron**
Task/Issue URL: https://app.asana.com/0/1206329551987282/1207736844779819/f
CC: @SabrinaTardio 

**Description**:

This PR refactors the view hierarchy to make the UI Dialog seem to update the same view and match the Android animation.
I also cleaned up a bit by moving/renaming some views.

https://github.com/duckduckgo/iOS/assets/1089358/cfdb15b1-021b-439d-815a-005b93ee4cda

**Steps to test this PR**:
1. Run the new Onboarding Intro flow
2. Check that the animation from the intro dialog to the browser comparison dialog looks nice and smooth.

**Definition of Done (Internal Only)**:

* [ ] Does this PR satisfy our [Definition of Done](https://app.asana.com/0/1202500774821704/1207634633537039/f)?

**Copy Testing**:

* [ ] Use of correct apostrophes in new copy, ie `’` rather than `'`

**Orientation Testing**:

* [ ] Portrait
* [ ] Landscape

**Device Testing**:

* [ ] iPhone SE (1st Gen)
* [ ] iPhone 8
* [ ] iPhone X
* [ ] iPhone 14 Pro
* [ ] iPad

**OS Testing**:

* [ ] iOS 15
* [ ] iOS 16
* [ ] iOS 17

**Theme Testing**:

* [ ] Light theme
* [ ] Dark theme

---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)

### CODE REVIEW

File: DuckDuckGo/OnboardingExperiment/OnboardingIntro/OnboardingView+BrowsersComparisonContent.swift [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import SwiftUI
+import DuckUI
+
+extension OnboardingView {
+
+    struct BrowsersComparisonContent: View {
```
Comment (alessandroboron) [inline DuckDuckGo/OnboardingExperiment/OnboardingIntro/OnboardingView+BrowsersComparisonContent.swift:25]: This was `DaxDialogBrowserComparisonView`. But now it’s “just” one of the content that we inject inside DaxDialog

File: DuckDuckGo/OnboardingExperiment/OnboardingIntro/OnboardingView+IntroDialogContent.swift [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import SwiftUI
+import DuckUI
+
+extension OnboardingView {
+
+    struct IntroDialogContent: View {
```
Comment (alessandroboron) [inline DuckDuckGo/OnboardingExperiment/OnboardingIntro/OnboardingView+IntroDialogContent.swift:25]: This was `DaxDialogIntroView`. But now it’s “just” one of the content that we inject inside DaxDialog

File: DuckDuckGo/OnboardingExperiment/OnboardingIntro/OnboardingView+Landing.swift [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import SwiftUI
+
+extension OnboardingView {
```
Comment (alessandroboron) [inline DuckDuckGo/OnboardingExperiment/OnboardingIntro/OnboardingView+Landing.swift:22]: This was existing already, I moved it out from OnboardingView to have a leaner file

File: DuckDuckGo/OnboardingExperiment/OnboardingIntro/OnboardingView.swift [hunk 4]
```diff
@@ -27,37 +27,63 @@ struct OnboardingView: View {
...
+                                introView
+                            case .browsersComparisonDialog:
+                                browsersComparisonView
+                            }
+                        }
+                    }
+                )
             }
+            .frame(width: geometry.size.width, alignment: .center)
             .offset(y: geometry.size.height * Metrics.dialogVerticalOffsetPercentage.build(v: verticalSizeClass, h: horizontalSizeClass))
```
Comment (alessandroboron) [inline DuckDuckGo/OnboardingExperiment/OnboardingIntro/OnboardingView.swift:84]: I used a Spacer initially. The issue is that the design for iPad has the dialogs “almost” centered vertically on screen. For this reason I had to leave the offset.

**Review by brindy — APPROVED** [review]
LGTM good job

**Pull request merged**